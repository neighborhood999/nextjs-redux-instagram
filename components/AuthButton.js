import React from 'react';
import PropTypes from 'prop-types';
import { compose, onlyUpdateForPropTypes, setPropTypes } from 'recompose';
import Link from 'next/link';
import { Button, Grid } from 'semantic-ui-react';

const AuthButton = ({ authStatus }) =>
  authStatus
    ? <Grid centered>
        <Grid.Row>
          <Button color="green">
            <Link href="#">
              <a style={{ color: '#FFFFFF', textDecoration: 'none' }}>
                Authentication
              </a>
            </Link>
          </Button>
        </Grid.Row>
      </Grid>
    : <Grid centered>
        <Grid.Row>
          <Button basic color="blue">
            <Link href="#">
              <a>Profile</a>
            </Link>
          </Button>
        </Grid.Row>
      </Grid>;

export default compose(
  onlyUpdateForPropTypes,
  setPropTypes({ authStatus: PropTypes.bool.isRequired })
)(AuthButton);
