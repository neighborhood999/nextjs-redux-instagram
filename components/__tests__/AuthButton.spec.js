import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import AuthButton from '../AuthButton';

describe('AuthButton Component', () => {
  test('AuthButton should be wrapped up in onlyUpdateForPropTypes', () => {
    expect(AuthButton.displayName).toBe('onlyUpdateForPropTypes(AuthButton)');
  });

  test('should be `Authentication` text when authStatus is `true`', () => {
    const component = mount(<AuthButton authStatus={true} />);
    expect(component.find('a').text()).toBe('Authentication');

    const tree = renderer.create(<AuthButton authStatus={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should be `Profile` text when authStatus is `false`', () => {
    const component = mount(<AuthButton authStatus={false} />);
    expect(component.find('a').text()).toBe('Profile');

    const tree = renderer.create(<AuthButton authStatus={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
