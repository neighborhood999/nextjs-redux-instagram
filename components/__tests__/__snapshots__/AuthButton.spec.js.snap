// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton Component should be \`Authentication\` text when authStatus is \`true\` 1`] = `
<div
  className="ui centered grid"
>
  <div
    className="row"
  >
    <button
      className="ui green button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <a
        href="https://api.instagram.com/oauth/authorize/?client_id=undefined&;redirect_uri=http://localhost:3000/&response_type=code&scope=public_content+follower_list"
        onClick={[Function]}
        style={
          Object {
            "color": "#FFFFFF",
            "textDecoration": "none",
          }
        }
      >
        Authentication
      </a>
    </button>
  </div>
</div>
`;

exports[`AuthButton Component should be \`Profile\` text when authStatus is \`false\` 1`] = `
<div
  className="ui centered grid"
>
  <div
    className="row"
  >
    <button
      className="ui blue basic button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <a
        href="/user-profile"
        onClick={[Function]}
      >
        Profile
      </a>
    </button>
  </div>
</div>
`;
